content/src/content/docs/docs/ai/planning-llm-interactions.mdx

```diff
@@ -0,0 +1,227 @@
+---
+title: Execution Planning
+description: Learn how to create structured execution plans for your LLM interactions
+sidebar:
+  order: 2
+---
+
+Imagine that we've refactored our `generateDadJoke` program from our [Getting Started](/docs/ai/getting-started/) guide. Now, instead of handling all errors internally, the code can **fail with domain-specific issues** like network interruptions or provider outages:
```

Comment:
> Now, instead of handling all errors internally, the code can fail with domain-specific issues like network interruptions or provider outages.

How does that happen? This caught me off guard. It might be worth adding a section to "Getting Started" that covers "Error Handling", showing how you can end up in this situation.

Reply:
I'm not sure how to show this other than just failing in the `generateDadJoke` program with specific errors. It's no different than [failing with an error in any other Effect program](https://effect.website/docs/error-management/expected-errors/).

In this case, I'm just saying "Ok, let's say the logic in `generateDadJoke` can now fail with these possible errors". I don't think it's worth introducing a contrived failure scenario in code unless you do. To me, the type declaration was enough.